File: src/androidTest/java/org/tasks/makers/RemoteGtaskListMaker.java
package org.tasks.makers;

import com.google.api.client.util.DateTime;
import com.google.api.services.tasks.model.TaskList;
import com.natpryce.makeiteasy.Instantiator;
import com.natpryce.makeiteasy.Property;
import com.natpryce.makeiteasy.PropertyLookup;
import com.natpryce.makeiteasy.PropertyValue;

import static com.natpryce.makeiteasy.Property.newProperty;
import static org.tasks.time.DateTimeUtils.currentTimeMillis;
import static org.tasks.makers.Maker.make;

public class RemoteGtaskListMaker {
    public static final Property<TaskList, String> REMOTE_ID = newProperty();
    public static final Property<TaskList, String> NAME = newProperty();

    public static TaskList newRemoteList(PropertyValue<? super TaskList, ?>... properties) {
        return make(instantiator, properties);
    }

    private static final Instantiator<TaskList> instantiator = new Instantiator<TaskList>() {
        @Override
        public TaskList instantiate(final PropertyLookup<TaskList> lookup) {
            return new TaskList()
                    .setId(lookup.valueOf(REMOTE_ID, "1"))
                    .setTitle(lookup.valueOf(NAME, "Default"))
                    .setUpdated(new DateTime(currentTimeMillis()));
        }
    };
}

File: app/src/test/java/org/tasks/makers/RemoteGtaskListMaker.kt
package org.tasks.makers

import com.google.api.client.util.DateTime
import com.google.api.services.tasks.model.TaskList
import com.natpryce.makeiteasy.Instantiator
import com.natpryce.makeiteasy.Property
import com.natpryce.makeiteasy.Property.newProperty
import com.natpryce.makeiteasy.PropertyLookup
import com.natpryce.makeiteasy.PropertyValue
import org.tasks.makers.Maker.make
import org.tasks.time.DateTimeUtils2.currentTimeMillis

object RemoteGtaskListMaker {
    val REMOTE_ID: Property<TaskList, String> = newProperty()
    val NAME: Property<TaskList, String> = newProperty()

    private val instantiator = Instantiator { lookup: PropertyLookup<TaskList> ->
        TaskList()
                .setId(lookup.valueOf(REMOTE_ID, "1"))
                .setTitle(lookup.valueOf(NAME, "Default"))
                .setUpdated(DateTime(currentTimeMillis()).toStringRfc3339())
    }

    fun newRemoteList(vararg properties: PropertyValue<in TaskList?, *>): TaskList {
        return make(instantiator, *properties)
    }
}